### PR TITLE
RSA benches

### PR DESCRIPTION
**Pull request by bgrieder**
Add RSA benches

### CODE REVIEW

File: crate/test_server/benches/benches.rs [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+    kms_rest_client: &KmsClient,
+    pk: &str,
+    cleartext: &[u8],
+    cryptographic_parameters: &CryptographicParameters,
+) -> Vec<u8> {
+    // Create the kmip query
+    let encrypt_request = build_encryption_request(
+        pk,
+        None,
+        cleartext.to_vec(),
```
Comment (tbrezot) [inline crate/test_server/benches/benches.rs:309]: Take ownership instead.

File: crate/test_server/benches/benches.rs [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+        None,
+        None,
+        Some(cryptographic_parameters.clone()),
```
Comment (tbrezot) [inline crate/test_server/benches/benches.rs:312]: Take ownership instead.

File: crate/test_server/benches/benches.rs [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+        ciphertext.to_vec(),
+        None,
+        None,
+        Some(cryptographic_parameters.clone()),
+    );
+
+    // Query the KMS with your kmip data and get the key pair ids
+    let decrypt_response = kms_rest_client.decrypt(decrypt_request).await.unwrap();
+
+    decrypt_response.data.unwrap().to_vec()
```
Comment (tbrezot) [inline crate/test_server/benches/benches.rs:341]: Maybe do not convert to vec to allow allocation.

**Review by tbrezot — COMMENTED** [review]
Nice, just some allocation that may possibly be moved out of the benched function.

File: crate/test_server/benches/benches.rs (hunk 2 above)
Comment (Manuthor) [reply crate/test_server/benches/benches.rs]: done in f29feec55cba8edbe1e80ea9f53823f1d4603c27

File: crate/test_server/benches/benches.rs (hunk 3 above)
Comment (Manuthor) [reply crate/test_server/benches/benches.rs]: done in f29feec55cba8edbe1e80ea9f53823f1d4603c27

File: crate/test_server/benches/benches.rs (hunk 1 above)
Comment (Manuthor) [reply crate/test_server/benches/benches.rs]: done in f29feec55cba8edbe1e80ea9f53823f1d4603c27

**Review by Manuthor — APPROVED** [review]

**Pull request merged**